### ISSUE TITLE
DIRECT_FP32_GEMV: a FALSIFIED optimization left as an env-gated branch that no gate executes

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#3915]
## Summary

`crates/aprender-serve/src/quantize/parallel_k.rs` contains an env-gated
alternate GEMV path:

```rust
// PMAT-305: Direct FP32 FALSIFIED (-17%, 25.4 vs 30.8 tok/s).
// Q8K + maddubs (32 muls/insn) beats FP32 fmadd (8 muls/insn).
let use_direct_fp32 = std::env::var("DIRECT_FP32_GEMV").as_deref() == Ok("1");
if use_direct_fp32 { ... return Ok(()); }
```

**It is not an untested feature. It is a measured-and-rejected experiment that
was left in the tree**, and nothing exercises it.

## Three facts, each verified

**1. It was falsified.** The code's own comment records the measurement:
`-17%, 25.4 vs 30.8 tok/s`. Q8K+maddubs beat it. It is not a fast path anyone is
waiting on.

**2. It has already caused a bug.** `contracts/apr-load-fail-closed-truncated-v1.yaml`
describes its own origin as *"an adversarial inference bug-hunt (root cause
behind a narrow DIRECT_FP32_GEMV panic)"*. So this branch has produced at least
one panic, and the fix was hardening a *different* code path.

**3. No gate we own has ever executed it.**

```
grep -rln 'DIRECT_FP32_GEMV' scripts/ .github/ Makefile
  (no matches)
```

References exist only in three source files, one contract, and docs. Nothing
sets the variable, so every CI run, every `make` tier, and every ladder run takes
the other branch.

## How it was found, which is the part worth keeping

While investigating #3899 I planted three separate fault-injection distortions in
that `par_chunks_mut` block to test a guard. **All three had zero effect**, and a
`panic!` probe at the function's entry *did* fire — so "the function is reached"
and "my edit does nothing" were simultaneously true. I briefly concluded, and
reported, that the CPU generate path was unlocatable. That was wrong: I was
editing a branch that is dead at runtime.

**Anyone doing fault injection, coverage work, or a bisect on the Q4_K CPU matvec
will hit the same wall**, and the failure mode is silent — the mutation applies,
the code compiles, the tests pass, and the conclusion drawn is about the code
when it is really about the probe.

## Why this is the dangerous member of its family

This is the third instance of one shape, and the three should be findable from
each other:

| instance | shape |
|---|---|
| #3809 / #3810 | a **test** nothing runs |
| #3902 | a test whose **failing configuration** nothing runs |
| **this** | a **code path** nothing runs |

The family is not "dark tests", it is **dark configurations**. This is the worst
member because it is switchable at runtime by an environment variable, on a path
no gate has ever measured — so the first person to set `DIRECT_FP32_GEMV=1` while
debugging gets a GEMV that was last validated before PMAT-305 and has already
panicked once.

## Suggested disposition

**Delete it**, rather than test it. A falsified optimization retained behind an
env var is pure liability: it cannot be a fast path (measured slower), it is not
covered, and it has a panic in its history. If it is kept for future
experimentation it needs at minimum a gate that exercises it, which is a strictly
worse trade than removing ~20 lines.

If deletion is refused, the alternative is a CI job that runs the CPU quant suite
with `DIRECT_FP32_GEMV=1` — otherwise this ticket recurs the next time someone
tries to mutate that block.

## Not claimed

* Not claimed the branch is currently wrong. It is unmeasured, which is the point.
* Not claimed it caused the #3899 divergence — it did not; #3899 is closed as a
  near-tie (0.243 logits against a 3.260 spread).

Found while investigating #3899. Related: #3902, #3809, PMAT-305.

**Comment by noahgift** [comment]
## Disposition: 0.70.0, first fold after the tag — DELETE endorsed on the merits

Milestoned 0.70.0 by the release cop's ruling. **The deletion is endorsed; only
the timing is deferred**, and the reason is sequencing rather than doubt — so
nobody re-litigates the merits later.

**Why not 0.69.1:** every change to the release branch invalidates the model
ladder receipts, which must carry the tagged sha. Each additional fold costs
**two ladder re-runs, roughly an hour each**. This is dead code that has been
dead for a while and appears in no judge row, so spending two card-hours to
remove it on the tag path is the wrong trade — and it is exactly the
"while we're here" change that turns a release into a week.

**What is settled and needs no re-deciding:**

* the path was **measured and rejected** (`-17%, 25.4 vs 30.8 tok/s`, in the
  code's own comment) — so "add a CI job to cover it" buys coverage for a known
  regression
* it has **already produced a panic**, and the remediation hardened a *different*
  code path (`contracts/apr-load-fail-closed-truncated-v1.yaml`)
* **no gate executes it** — zero references in `scripts/`, `.github/`, `Makefile`
* removing ~20 lines beats adding a CI job to exercise a slower path

**One addition worth recording for whoever picks this up.** The falsification
comment sits **three lines above** the `env::var` check. I edited inside that
`if` block three separate times without seeing it. That is not carelessness so
much as what a focused edit does to peripheral vision, and it is the practical
argument for deletion over retention: a branch whose "do not use this" is a
comment will keep costing someone the same hour.

Found while investigating #3899 (closed: near-tie, not a CUDA defect). Family:
#3902, #3809.

**Comment by noahgift** [comment]
milestone → 0.73.0 per docs/roadmaps/milestone-triage-2026-09-23.md (#4024), operator approval 2026-09-23 (relayed by aprender-cf). Reason: a falsified optimization left as dead env-gated code: debt ratchet (#3997), slice 4
